compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -630,33 +631,54 @@ object ProtoTypes {
       approxAnd
     case tp: OrType =>
       def approxOr = {
-        val tp1a = wildApprox(tp.tp1, theMap, seen)
-        val tp2a = wildApprox(tp.tp2, theMap, seen)
+        val tp1a = wildApprox(tp.tp1, theMap, seen, internal)
+        val tp2a = wildApprox(tp.tp2, theMap, seen, internal)
         if (tp1a.isInstanceOf[WildcardType] || tp2a.isInstanceOf[WildcardType])
           WildcardType(tp1a.bounds | tp2a.bounds)
         else
           tp.derivedOrType(tp1a, tp2a)
       }
       approxOr
     case tp: SelectionProto =>
-      tp.derivedSelectionProto(tp.name, wildApprox(tp.memberProto, theMap, seen), NoViewsAllowed)
+      tp.derivedSelectionProto(tp.name, wildApprox(tp.memberProto, theMap, seen, internal), NoViewsAllowed)
     case tp: ViewProto =>
       tp.derivedViewProto(
-          wildApprox(tp.argType, theMap, seen),
-          wildApprox(tp.resultType, theMap, seen))
+          wildApprox(tp.argType, theMap, seen, internal),
+          wildApprox(tp.resultType, theMap, seen, internal))
     case  _: ThisType | _: BoundType => // default case, inlined for speed
       tp
+    case tl: TypeLambda =>
+      val internal1 = internal + tl
+      val paramInfos = tl.paramInfos
+      val paramInfos1 = tl.paramInfos.mapConserve {
+        case tb @ TypeBounds(lo, hi) =>
+          tb.derivedTypeBounds(
+            wildApprox(lo, theMap, seen, internal1),
+            wildApprox(hi, theMap, seen, internal1)
+          )
+      }
+      val res = tl.resultType
+      val res1 = wildApprox(res, theMap, seen, internal1)
+      if ((res eq res1) && (paramInfos eq paramInfos1)) tl
+      else {
+        def substBounds(tl1: TypeLambda)(tb: TypeBounds): TypeBounds =
+          tb.derivedTypeBounds(tb.lo.subst(tl, tl1), tb.hi.subst(tl, tl1))
+        tl match {
+          case _: HKTypeLambda => HKTypeLambda(tl.paramNames)(tl1 => paramInfos.map(substBounds(tl1)), tl1 => res1.subst(tl, tl1))
+          case _: PolyType     => PolyType(tl.paramNames)(tl1 => paramInfos.map(substBounds(tl1)), tl1 => res1.subst(tl, tl1))
+        }
+      }
```

Comment:
I *think* all this machinery is already provided by `derivedLambdaType`. Can you try this instead as the case for `TypeLambda`?
```scala
  val internal1 = internal + tl
  val paramInfos1 = tl.paramInfos.mapConserve {
    case tb @ TypeBounds(lo, hi) =>
          tb.derivedTypeBounds(
            wildApprox(lo, theMap, seen, internal1),
            wildApprox(hi, theMap, seen, internal1)
          )
      }
  val res1 = wildApprox(tl.resultType, theMap, seen, internal1)
  tl.derivedLambdaType(paramInfos = paramInfos1, resType = res1)
```

Reply:
Damn. That's what I expected, and I looked for it, but somehow managed to miss `newLikeThis` and then ended up rewriting it inline :-(

Reply:
It can even be simplified further, it seems. See 85c641f.